src/logic/basic.lean

```diff
@@ -152,6 +152,10 @@ assume ⟨h⟩, h.elim
 @[simp] theorem exists_pempty {P : pempty → Prop} : (∃ x : pempty, P x) ↔ false :=
 ⟨λ h, by { cases h with w, cases w }, false.elim⟩
 
+lemma congr_heq {α β γ : Sort*} {f : α → γ} {g : β → γ} {x : α} {y : β} (h₁ : f == g)
+  (h₂ : x == y) : f x = g y :=
+by { have := type_eq_of_heq h₂, subst this, rw [eq_of_heq h₁, eq_of_heq h₂] }
```

Comment:
There might be a shorter proof with pattern matching if you move everything but the types after the colon

Reply:
None of the arguments are equalities though, so I don't see how that would help.

Reply:
Pattern matching should work on heq too

Reply:
Hmm, seems I'm wrong about that

Reply:
Still quite golfable:
```suggestion
by { cases h₂, cases h₁, refl }
```